### PR TITLE
Transfer query modules from MAGE to Memgraph

### PR DESCRIPTION
**Pull request by mpintaric55334**
[master < Task] PR
- [x] Check, and update documentation if necessary
- [x] Provide the full content or a guide for the final git message

Transfer of mgps, schema and convert QM from MAGE to Memgraph

To keep docs changelog up to date, one more thing to do:
- [x] Write a release note here, including added/changed clauses
- [x] Tag someone from docs team in the comments

### CODE REVIEW

File: tests/e2e/query_modules/workloads.yaml [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+      validation_queries: []
+
+disk_cluster: &disk_cluster
+  cluster:
+    main:
+      args: *args
+      log_file: "query_modules-e2e.log"
+      setup_queries:
+        - "STORAGE MODE ON_DISK_TRANSACTIONAL"
+      validation_queries: []
```
Comment (antoniofilipovic) [inline tests/e2e/query_modules/workloads.yaml:22]: You don't need disk cluster here as it is not used

File: tests/e2e/runner.py [hunk 2]
```diff
@@ -56,6 +56,10 @@ def run(args):
         def cleanup():
             interactive_mg_runner.stop_all()
 
+        if "presetworkload" in workload:
```
Comment (antoniofilipovic) [inline tests/e2e/runner.py:59]: please rename this variable

File: tests/e2e/query_modules/query_modules.py [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+    )
+    assert (result) == [":`Dog`", ["Dog"], "name", ["String"], True]
+
+    result = list(
+        execute_and_fetch_all(
+            cursor,
+            f"CALL libschema.node_type_properties() YIELD nodeType, nodeLabels, propertyName, propertyTypes , mandatory RETURN nodeType, nodeLabels, propertyName, propertyTypes , mandatory ORDER BY propertyName, nodeLabels[0];",
+        )[3]
+    )
+    assert (result) == [":`Dog`", ["Dog"], "owner", ["String"], True]
```
Comment (antoniofilipovic) [inline tests/e2e/query_modules/query_modules.py:76]: Does it make sense to move this part to new file as this is just test for `node_type_properties`

File: tests/e2e/query_modules/query_modules.py [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+        cursor,
+        "CREATE (d:Dog {name: 'Rex', owner: 'Carl'})-[l:LOVES]->(a:Activity {name: 'Running', location: 'Zadar'})",
+    )
+    result = list(
+        execute_and_fetch_all(
+            cursor,
+            f"CALL libschema.rel_type_properties() YIELD relType,propertyName, propertyTypes , mandatory RETURN relType, propertyName, propertyTypes , mandatory;",
+        )[0]
+    )
+    assert (result) == [":`LOVES`", "", "", False]
```
Comment (antoniofilipovic) [inline tests/e2e/query_modules/query_modules.py:91]: Same comment about this one

File: tests/e2e/query_modules/query_modules.py [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+def test_convert_list1():
+    cursor = connect().cursor()
+    result = execute_and_fetch_all(cursor, f"RETURN convert.str2object('[2, 4, 8, [2]]') AS result;")[0][0]
+    assert (result) == [2, 4, 8, [2]]
+
+
+def test_convert_list_wrong():
+    cursor = connect().cursor()
+    result = execute_and_fetch_all(cursor, f"RETURN convert.str2object('[2, 4, 8, [2]]') AS result;")[0][0]
+    assert (result) != [3, 4, 8, [2]]
```
Comment (antoniofilipovic) [inline tests/e2e/query_modules/query_modules.py:27]: Also, maybe we can move this to other file, so we have tests split in different groups

**Review by antoniofilipovic — CHANGES_REQUESTED** [review]
Few changes, otherwise looks good

File: tests/e2e/query_modules/query_modules.py (hunk 4 above)
Comment (mpintaric55334) [reply tests/e2e/query_modules/query_modules.py:91]: Ill move em in schema test

File: tests/e2e/query_modules/query_modules.py (hunk 5 above)
Comment (mpintaric55334) [reply tests/e2e/query_modules/query_modules.py:27]: I dont think that makes sense, I think we should do, one file, one QM. If we separate one QM into multiple files for each test we are looking for big chaos in future

**Comment by vpavicic** [comment]
Note to myself:
We need to move the information about the procedures from the advanced-algorithms/available algorithms to the respective pages...
Maybe we create a Procedures sub-page within Querying and move the information there, then link from important and relevant pages?

**Review by antoniofilipovic — APPROVED** [review]

**Review by gitbuda — APPROVED** [review]

**Pull request merged**